[Chen Shun] IP

csheun opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

Commit:
Add Gradle support

Commit:
Completed Level-1: Greet, Echo, Exit

Commit:
Completed Level-2:Add, List

Commit:
Completed Level-3: Mark as Done

Commit:
Completed Level-4: ToDo, Event, Deadline

Commit:
Completed A-TextUiTesting

Commit:
Completed Level-5: Handle Errors

Commit:
Completed Level-6: Delete

Commit:
Edit week 1 code

LianGuoYang commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+    private String line  = "___________________________________________________";
+    private String exit = "Bye. Hope to see you again soon!";
+    Scanner sc = new Scanner(System.in);
+    ArrayList<Task> tasks;
```
Should tasks be private instead of public ?

LianGuoYang commented on src/main/java/PonyException.java:
```diff
@@ -0,0 +1,33 @@
+        }
+    }
+
+}
```
I like where you have multiple exceptions to account for different errors.

LianGuoYang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    private String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // mark done task with X
```
Perhaps you can move the comment to the top.

LianGuoYang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+        return this.getStatusIcon() + " " + this.description;
+    }
+
+    //...
```
Should this be removed?

LianGuoYang commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,12 @@
+public class ToDo extends Task {
+
```
Extra line break here, should this be removed?

LianGuoYang commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+            } else {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < this.tasks.size(); i++) {
+                    int sn = i + 1;
```
Maybe a intuitive name for variable sn?

LianGuoYang commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+        myPony.initialise();
+        myPony.run();
+    }
+}
```
I like where you throw exceptions for each commands and how you added comments for each command which made reading your code easier.

LianGuoYang commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+public class Pony {
```
Perhaps a line break to separate line 2 and line 3?

LianGuoYang commented on src/main/java/PonyException.java:
```diff
@@ -0,0 +1,33 @@
+import java.lang.RuntimeException;
+public class PonyException {
```
Maybe a line break to separate both lines?

Commit:
Fix bug in creating event

LianGuoYang reviewed (commented):
LGTM. Just minor things to fix.

april-anh commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+                run();
+            }
+        } else if (action.equals("todo")) {
+            // tod0 expects task description in command[1]
```
You can use "to-do" to prevent the text to be marked green.

april-anh commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+    private void processCommand(String[] command) {
+        int commandSize = command.length;
+        String action = command[0];
+        if (action.equals("list")) {
```
Maybe using `switch` instead of `if-else` can make your code neater?

april-anh commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+            } catch (NumberFormatException e) {
+                System.out.println(new PonyException.taskInputError().getMessage());
+            } finally {
+                run();
```
I like how you carefully clean up your program after handling exceptions. :+1:

april-anh reviewed (commented):
Detailed comments make your code very easy to understand.

tensaida commented on src/main/java/PonyException.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    //Task to mark not provided
+    public static class taskMissingError extends RuntimeException {
```
You could make these Errors extend PonyException instead

tensaida commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+    }
+
+    public static void main(String[] args) {
+        Pony myPony = new Pony();
```
Since only one `Pony` object is created, why not make everything `static` instead?

tensaida commented on src/main/java/Pony.java:
```diff
@@ -0,0 +1,195 @@
+                run();
+            }
+        } else if (action.equals("unmark")) {
+            // unmark expects input in command[1] -> which task to unmark
```
good comment 👍🏾

tensaida reviewed (commented):
Good job! All the best for the upcoming weeks

Commit:
Fix Storage

Commit:
implement save

Commit:
implement time

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Improve OOP

Commit:
Add packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Add Gradle

Commit:
Add JUnit Tests

Commit:
Add shadowJar setting

Commit:
Implement Find command

Commit:
Add JavaDocs

Commit:
Improve code quality

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Add GUI

Commit:
Add assertions

Commit:
Improve code quality

csheun committed:
Merge pull request #2 from ciaoosuuu/branch-A-Assertions

Add assertions

csheun committed:
Merge pull request #3 from ciaoosuuu/branch-A-CodeQuality

Improve code quality

Commit:
Add detect duplicates function

Commit:
Add personality to pony

Response from pony is not friendly to user.

Change some of the pony's response to a more friendly tone.

Commit:
Find command: make matching case insensitive

Find command is case sensitive.

A case insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's update the search algorithm by converting all
strings to lowercase and find matches in lowercase.

Commit:
Edit JavaDoc and improve code quality

Commit:
Fix bug in todo accepting blank messages

Commit:
Add Ui and readme

Commit:
Update README

Commit:
Update README and fix typo

Commit:
Update README

Commit:
Update README

Commit:
Update README

Commit:
Update README
